### DISCUSSION TITLE
Rotating Layers with Leaflet Rotate

### DISCUSSION THREAD

**Discussion by cubecleveland** [geoman-io/leaflet-geoman#1097] [Questions & Help]
Hello,
Started using the Leaflet rotate plugin to rotate the bearing of the map.
The plugin Works great but stated to see some weird behaiour when integrating editing on the rotates layer.

Here is a jsfiddle having the 2 libs together
https://jsfiddle.net/kua1rs5v/

Issus occur especially when drawing a rectangle as it seems the algorithm atatches itslef to the current tile, is there any way that can be fixed or taken into consideration ?

As of now, we have limited the drawing of rectangles but it would be great if u can give us some guidance of maybe fixing this issue. Issue is also when rotating a line.
Seems like the only shape that work with a rotated tile layer is polygon.

thanks again!

**Comment by Falke-Design** [comment]
The rotate branch of Leaflet is not ready for use and is also not maintained (1105 commits behinde the master). We will not support map rotation.

The rectangle "rotation" while drawing and editing is done here: 
https://github.com/geoman-io/leaflet-geoman/blob/1fdc918fa39ffa84327fdf639fa75865168f716d/src/js/L.PM.Utils.js#L191

And the rotation of layers is done here:
https://github.com/geoman-io/leaflet-geoman/blob/1fdc918fa39ffa84327fdf639fa75865168f716d/src/js/Mixins/Rotating.js#L112
